src/main/DynamicConfig.ts

```diff
@@ -299,13 +299,46 @@ export class DynamicConfig implements IDynamicConfig {
     }
 
     public async getRemoteValue<T>(key: string, type?: ObjectType): Promise<T> {
-        return this.source(key).then((source: ISource) => {
-            if (source.key !== undefined) {
-                return this.getValueFromResolver<T>(source.key, 'remote', type)
-            } else {
-                throw new errors.ResolverUnavailable(key)
-            }
-        })
+        const source = await this.source(key) // get source for key
+        const { key: sourceKey } = source
+        if (!sourceKey) {
+            throw new errors.ResolverUnavailable(key) // throw if key is undefined
+        }
+
+        const resolvedConfig = await this.getConfig() // get the current config (the cached version, that is)
+
+        const value = await this.getValueFromResolver<T>(
+            sourceKey,
+            'remote',
+            type,
+        ) // get new remote value for key
+
+        if (value === null) {
+            throw new errors.ResolverUnavailable(key) // if it doesn't exist, throw
+        }
+
+        const normalizedKey: string = Utils.normalizePath(key) // normalize/format key path
+
+        /*
+         build the normalized key path for the refreshed value to live at.
+         *note*: this is **required** to format the new value into a shape consumable by `setConfig()`.
+         */
+        const builtValue = ConfigBuilder.buildBaseConfigValue(source, value)
+
+        await this.setConfig(
+            ConfigUtils.setValueForKey(
+                normalizedKey,
+                builtValue,
+                resolvedConfig,
+                true,
+            ) as IRootConfigValue,
+        ) // ship the formatted value to be set in the config
+
+        /*
+        return the value we got from the remote source. We _could_ also do a `get()` against the updated config to be sure the value was updated.
+        I can't be sure how much latency it'd add, but I'm inclined to do it anyways instead of doing an optimistic update.
+        */
+        return value
```

Comment:
/*
        return the value we got from the remote source. We _could_ also do a `get()` against the updated config to be sure the value was updated.
        I can't be sure how much latency it'd add, but I'm inclined to do it anyways instead of doing an optimistic update.
        */